Library/Homebrew/requirement.rb

```diff
@@ -92,8 +99,15 @@ def satisfied_result_parent
     parent
   end
 
-  # Overriding {#modify_build_environment} is unsupported.
-  # Pass a block to the env DSL method instead.
+  # Pass a block to the env DSL method instead of overriding.
+  sig(:final) {
+    params(
+      env:          T.nilable(String),
+      cc:           T.nilable(String),
+      build_bottle: T::Boolean,
+      bottle_arch:  T.nilable(String),
```

Comment:
params are based on https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/ENV.rb#L36-L39 (though I noticed that `build_bottle` is `T.nilable(T::Boolean)` in some places)

Reply:
I think `T::Boolean` is preferable in most cases. We should adjust the call sites to avoid `T.nilable(T::Boolean)` being passed where this is still the case.